d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -175,6 +175,10 @@ public XdsClientImpl(Node node,
   @Override
   public void watchXdsResource(String resourceName, ResourceWatcher watcher)
   {
+    if (_shutdown) {
```

Comment:
Question for reviewers: would it be more idiomatic to use `_shutdown` or `_executorService.isShutdown` here? They should be the same based on the implementation of `shutdown()`

Reply:
They are different. _shutdown won't be set to true if _executorService is already shut down, will reject execution at line 312.
Can you rename _shutdown to _isXdsStreamShutdown?

At line 312, it also needs to call checkShutdownAndExecute instead of directly calling .execute. If the executor is already shutdown for some reason (not sure if it really happens in edge case), we should still run the function in a new thread, just not to be executed in the executor.

Reply:
Makes sense! For line 312, would it be better to just always run it outside the executor? I guess we would want a new thread to make it async but we could just always spawn a new thread for the shutdown 🤔

Reply:
it's the executor is not shutdown, no need to spawn a new thread right? But in general, yeah, we need to make sure shutdown code is always executed and need to be async (don't want to be blocking in case ads stream hangs somehow).

Reply:
Sounds good; I've created a new executor for this just for simplicity, but let me know if it doesn't make sense